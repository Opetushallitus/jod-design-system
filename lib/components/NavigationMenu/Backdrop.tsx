import React from 'react';

export interface BackdropProps {
  children: React.ReactNode;
  dialogRef: React.RefObject<HTMLDialogElement | null>;
  onClose: () => void;
}

export const Backdrop = ({ children, dialogRef, onClose }: BackdropProps) => {
  const handleKeyDown = (event: React.KeyboardEvent) => {
    if (event.key === 'Escape') {
      onClose();
    }
  };

  const handleClickOutside = (event: React.MouseEvent) => {
    if (dialogRef.current && event.target === dialogRef.current) {
      onClose();
    }
  };

  return (
    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
    <dialog
      ref={dialogRef}
      className="ds:-z-1 ds:flex ds:backdrop:bg-black/30 ds:w-[370px] ds:max-w-full ds:max-h-screen ds:min-h-full ds:overflow-auto ds:overscroll-contain ds:hyphens-auto"
      onKeyDown={handleKeyDown}
      onClick={handleClickOutside}
    >
      {children}
    </dialog>
  );
};
