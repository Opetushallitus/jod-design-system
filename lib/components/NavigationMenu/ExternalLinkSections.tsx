import { RxExternalLink } from 'react-icons/rx';
import { Placeholder } from './Placeholder';

export interface LinkItem {
  label: string;
  url: string;
  description: string;
  accentColor: string;
}

export interface ExternalLinkSection {
  title: string;
  linkItems: LinkItem[];
}

const ExternalLinkItem = ({ item }: { item: LinkItem }) => {
  return (
    <li
      key={item.label}
      className="ds:border-l-[8px] ds:text-button-md ds:flex"
      style={{ borderColor: item.accentColor }}
    >
      <a
        href={item.url}
        target="_blank"
        rel="noopener noreferrer"
        className="ds:pl-5 ds:flex ds:flex-row ds:flex-1 ds:space-between ds:focus:outline-accent"
      >
        <div className="ds:flex ds:flex-col ds:flex-1 ds:gap-3 ds:py-3 ds:group">
          <div className="ds:flex ds:flex-row">
            <span className="ds:flex ds:flex-1 ds:group-hover:underline">{item.label}</span>
            <RxExternalLink size={24} role="presentation" />
          </div>
          <span className="ds:text-body-sm">{item.description}</span>
        </div>
      </a>
      <Placeholder />
    </li>
  );
};

export const ExternalLinkSections = ({ sections }: { sections: ExternalLinkSection[] }) => {
  return (
    <div>
      {sections.map((section) => (
        <div key={section.title}>
          <span className="ds:text-body-sm ds:mb-5 ds:mt-2 ds:flex">{section.title}</span>
          <ul className="ds:gap-3 ds:flex ds:flex-col">
            {section.linkItems.map((item) => (
              <ExternalLinkItem key={item.label} item={item} />
            ))}
          </ul>
        </div>
      ))}
    </div>
  );
};
