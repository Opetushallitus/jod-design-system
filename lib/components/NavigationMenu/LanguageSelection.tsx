import { MdCheck } from 'react-icons/md';
import { Placeholder } from './Placeholder';
import { LinkComponent } from './types';

export interface LanguageSelectionItem {
  label: string;
  value: string;
  linkComponent: React.ComponentType<LinkComponent>;
}

export interface NavigationMenuLanguageSelectionProps {
  /** Language selection items */
  items: LanguageSelectionItem[];
  /** Selected language */
  selected: string;
}

export const LanguageSelection = ({ items, selected }: NavigationMenuLanguageSelectionProps) => {
  return (
    <div className="ds:flex ds:flex-row">
      <ul aria-label="Language selection" className="ds:flex ds:flex-1 ds:flex-col">
        {items.map((item: LanguageSelectionItem) => (
          <li key={item.label} className="ds:flex ds:flex-1 ds:min-h-[40px] ds:items-center ds:hover:underline">
            <item.linkComponent
              className="ds:flex ds:flex-row ds:flex-1 ds:items-center ds:focus:outline-accent"
              aria-current={selected === item.value}
            >
              <span className="ds:size-[40px] ds:flex ds:justify-center ds:items-center">
                {selected === item.value && <MdCheck size={24} />}
              </span>
              <span className="ds:text-button-md">{item.label}</span>
            </item.linkComponent>
          </li>
        ))}
      </ul>
      <Placeholder />
    </div>
  );
};
