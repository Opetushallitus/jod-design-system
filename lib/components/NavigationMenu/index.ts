export { ExternalLinkSections, type ExternalLinkSection, type LinkItem } from './components/ExternalLinkSections';
export {
  LanguageSelection,
  type LanguageSelectionItem,
  type NavigationMenuLanguageSelectionProps,
} from './components/LanguageSelection';
export { MenuList, type MenuItem, type MenuListProps } from './components/MenuList';
export { NavigationMenu, type NavigationMenuProps } from './NavigationMenu';
export { type LinkComponent } from './types';
