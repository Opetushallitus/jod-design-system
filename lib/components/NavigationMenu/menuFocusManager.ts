const focusHistory: string[] = [];

export const addToFocusHistory = (id: string) => {
  focusHistory.push(id);
};

export const returnFocus = () => {
  if (focusHistory.length > 0) {
    const elementIdToFocus = focusHistory.pop();
    if (elementIdToFocus) {
      const button = document.querySelector(`[data-list-id="${elementIdToFocus}"] button`);
      if (button && button instanceof HTMLButtonElement) {
        setTimeout(() => {
          button.focus();
        }, 0);
      }
      return true;
    }
  }
  return false;
};

export const menuFocusManager = {
  addToFocusHistory,
  returnFocus,
};
