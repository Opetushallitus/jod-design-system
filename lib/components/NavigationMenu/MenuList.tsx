import React from 'react';
import { MdArrowBackIos, MdArrowForwardIos, MdHome } from 'react-icons/md';
import { menuFocusManager } from './menuFocusManager';
import { MenuSeparator } from './MenuSeparator';
import { Placeholder } from './Placeholder';
import { LinkComponent } from './types';

const FrontPageLink = ({
  label,
  icon = <MdHome size={24} />,
  component: Component,
}: {
  label: string;
  icon?: React.ReactNode;
  component: React.ComponentType<LinkComponent>;
}) => {
  return (
    <Component className="ds:flex ds:flex-row ds:flex-1 ds:gap-3 ds:p-3 ds:cursor-pointer ds:group ds:focus:outline-accent">
      {icon}
      <span className="ds:text-button-md ds:group-hover:underline">{label}</span>
    </Component>
  );
};

export interface MenuItem {
  label: string;
  LinkComponent?: React.ComponentType<LinkComponent>;
  childItems?: MenuItem[];
  selected?: boolean;
}

type MenuListItemProps = {
  accentColor: string;
  accentColorText?: string;
  onSubmenuClick?: () => void;
  openSubMenuLabel: string;
} & MenuItem;

const MenuListItem = ({
  label,
  selected,
  childItems,
  accentColor,
  accentColorText,
  LinkComponent,
  onSubmenuClick,
  openSubMenuLabel,
  ...rest
}: MenuListItemProps) => {
  return (
    <li className="ds:flex ds:flex-row ds:space-between ds:min-h-[40px] ds:gap-2" {...rest}>
      {LinkComponent ? (
        <LinkComponent className="ds:flex-1 ds:flex ds:focus:outline-accent" aria-current={selected}>
          <span
            className="ds:flex ds:items-center ds:text-button-md ds:flex-1 ds:p-3 ds:rounded ds:cursor-pointer ds:hover:underline"
            style={{ backgroundColor: selected ? accentColor : 'transparent' }}
          >
            {label}
          </span>
        </LinkComponent>
      ) : (
        <span
          className="ds:flex ds:items-center ds:text-[14px] ds:leading-[20px] ds:flex-1 ds:p-3"
          style={{ color: accentColorText ?? accentColor }}
        >
          {label}
        </span>
      )}
      {childItems && childItems.length > 0 ? (
        <button
          aria-label={openSubMenuLabel}
          className="ds:rounded ds:flex ds:items-center ds:justify-center ds:cursor-pointer ds:bg-bg-gray-2 ds:size-[40px] ds:text-black ds:hover:text-accent ds:focus:text-accent"
          onClick={onSubmenuClick}
        >
          <MdArrowForwardIos size={24} />
        </button>
      ) : (
        <Placeholder />
      )}
    </li>
  );
};

export interface MenuListProps {
  menuItems: MenuItem[];
  accentColor: string;
  accentColorText?: string;
  frontPageLinkLabel: string;
  frontPageIcon?: React.ReactNode;
  FrontPageLinkComponent: React.ComponentType<LinkComponent>;
  onNestedMenuOpen: (insideNestedMenu: boolean) => void;
  backLabel: string;
  openSubMenuLabel: string;
}

export const MenuList = ({
  menuItems,
  accentColor,
  accentColorText,
  frontPageLinkLabel,
  frontPageIcon,
  FrontPageLinkComponent,
  onNestedMenuOpen,
  backLabel,
  openSubMenuLabel,
}: MenuListProps) => {
  const [historyMenuItems, setHistoryMenuItems] = React.useState<MenuItem[][]>([]);
  const [shouldReturnFocus, setShouldReturnFocus] = React.useState(false);

  /** Going to nested menu */
  const onSubmenuItemClickHandler = (item: MenuItem) => {
    if (item.childItems && item.childItems.length > 0) {
      menuFocusManager.addToFocusHistory(item.label);

      setHistoryMenuItems((prev) => {
        const childItems = item.childItems || [];
        return [...prev, childItems];
      });
    }
  };
  /** Back to previous menu level */
  const onNestedMenuCloseHandler = () => {
    setHistoryMenuItems((prev) => {
      if (prev.length <= 1) {
        return [];
      }
      return prev.slice(0, -1);
    });
    setShouldReturnFocus(true);
  };

  /** MenuItems to show to user at the moment */
  const currentMenuItems = React.useMemo(() => {
    if (historyMenuItems && historyMenuItems.length > 0) {
      const lastItem = historyMenuItems[historyMenuItems.length - 1];
      return lastItem;
    }
    return menuItems;
  }, [menuItems, historyMenuItems]);

  React.useEffect(() => {
    if (historyMenuItems && historyMenuItems.length > 0) {
      onNestedMenuOpen(true);
    } else {
      onNestedMenuOpen(false);
    }
  }, [onNestedMenuOpen, historyMenuItems]);

  React.useEffect(() => {
    if (shouldReturnFocus) {
      menuFocusManager.returnFocus();
      setShouldReturnFocus(false);
    }
  }, [shouldReturnFocus]);

  const HomeButton = React.useCallback(() => {
    return (
      <div className="ds:flex ds:flex-row">
        {historyMenuItems && historyMenuItems.length > 0 ? (
          <button
            className="ds:flex ds:flex-row ds:flex-1 ds:gap-3 ds:p-3 ds:cursor-pointer ds:group ds:focus:outline-accent"
            onClick={onNestedMenuCloseHandler}
          >
            <MdArrowBackIos size={24} />
            <span className="ds:text-button-md ds:group-hover:underline">{backLabel}</span>
          </button>
        ) : (
          <FrontPageLink label={frontPageLinkLabel} icon={frontPageIcon} component={FrontPageLinkComponent} />
        )}
        <Placeholder />
      </div>
    );
  }, [historyMenuItems, backLabel, frontPageLinkLabel, frontPageIcon, FrontPageLinkComponent]);

  return (
    <>
      <HomeButton />
      <MenuSeparator />
      <div className="ds:border-l-[8px]" style={{ borderColor: accentColor }}>
        <ul className="ds:ml-3 ds:gap-2 ds:flex ds:flex-col">
          {currentMenuItems.map((item) => (
            <MenuListItem
              data-list-id={item.label}
              key={item.label}
              label={item.label}
              selected={item.selected}
              childItems={item.childItems}
              LinkComponent={item.LinkComponent}
              accentColor={accentColor}
              accentColorText={accentColorText}
              onSubmenuClick={() => onSubmenuItemClickHandler(item)}
              openSubMenuLabel={openSubMenuLabel}
            />
          ))}
        </ul>
      </div>
    </>
  );
};
