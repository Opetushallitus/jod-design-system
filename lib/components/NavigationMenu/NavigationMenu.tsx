import React from 'react';
import { ServiceVariantProvider } from '../../hooks/useServiceVariant/ServiceVariantProvider';
import { useServiceVariant } from '../../hooks/useServiceVariant/useServiceVariant';
import { JodClose } from '../../icons';
import { getFocusOutlineClassForService, type ServiceVariant } from '../../utils';
import { Backdrop } from './components/Backdrop';
import { ExternalLinkSection, ExternalLinkSections } from './components/ExternalLinkSections';
import { LanguageSelection, LanguageSelectionItem } from './components/LanguageSelection';
import { MenuList, type MenuSection } from './components/MenuList';
import { MenuSeparator } from './components/MenuSeparator';
import { PortalLink } from './components/PortalLink';
import type { LinkComponent } from './types';

const CloseMenuButton = ({ onClick, ariaCloseMenu }: { onClick: () => void; ariaCloseMenu: string }) => {
  const serviceVariant = useServiceVariant();
  return (
    <div className="ds:flex ds:justify-end">
      <button
        className={`ds:cursor-pointer ds:focus:outline-accent ds:p-3 ${getFocusOutlineClassForService(serviceVariant)}`}
        onClick={onClick}
        aria-label={ariaCloseMenu}
      >
        <JodClose size={24} />
      </button>
    </div>
  );
};

export interface NavigationMenuProps {
  /** Function to emit onClose event for user when the menu is closed */
  onClose: () => void;
  /** Open state of the NavigationMenu */
  open: boolean;
  /** Text for screenreader to have on the close menu button */
  ariaCloseMenu: string;
  /** Text for link that brings user to the front page */
  portalLinkLabel: string;
  /** Icon for the front page link */
  portalIcon?: React.ReactNode;
  /** Link component to bring user to front page */
  PortalLinkComponent: React.ComponentType<LinkComponent>;
  /** Menu items. Items can have children */
  menuSection: MenuSection;
  /** Label for button to open submenu of menu item */
  openSubMenuLabel: string;
  /** External link sections */
  externalLinkSections?: ExternalLinkSection[];
  /** Language selection items */
  languageSelectionItems?: LanguageSelectionItem[];
  /** Language selection title */
  languageSelectionTitle: string;
  /** Selected language */
  selectedLanguage: string;
  /** Extra section to be displayed at the end of the menu before the language selection */
  extraSection?: React.ReactNode;
  /** Service variant for color scheme (yksilÃ¶, ohjaaja etc.) */
  serviceVariant: ServiceVariant;
  /** Test id for querying in tests */
  dataTestId?: string;
}
export const NavigationMenu = ({
  onClose,
  open,
  ariaCloseMenu,
  portalLinkLabel,
  portalIcon,
  PortalLinkComponent,
  menuSection,
  openSubMenuLabel,
  externalLinkSections,
  languageSelectionItems,
  selectedLanguage,
  extraSection,
  languageSelectionTitle,
  serviceVariant,
  dataTestId,
}: NavigationMenuProps) => {
  const dialogRef = React.useRef<HTMLDialogElement>(null);

  React.useEffect(() => {
    const dialog = dialogRef.current;
    if (!dialog) {
      return;
    }

    if (open) {
      dialog.showModal();
    } else {
      dialog.close();
      onClose();
    }
  }, [dialogRef, open, onClose]);

  return open ? (
    <ServiceVariantProvider value={serviceVariant}>
      <Backdrop dialogRef={dialogRef} onClose={onClose} dataTestId={dataTestId ? `${dataTestId}-backdrop` : undefined}>
        <nav
          className="ds:bg-white ds:flex ds:flex-col ds:z-100 ds:flex-1"
          data-testid={dataTestId ? `${dataTestId}-root` : undefined}
        >
          <div
            className="ds:px-3 ds:flex ds:flex-col ds:overflow-y-auto ds:flex-grow"
            data-testid={dataTestId ? `${dataTestId}-body` : undefined}
          >
            <div className="ds:flex ds:items-center ds:justify-end ds:my-6">
              <CloseMenuButton onClick={onClose} ariaCloseMenu={ariaCloseMenu} />
            </div>
            {portalLinkLabel && PortalLinkComponent ? (
              <>
                <PortalLink label={portalLinkLabel} icon={portalIcon} component={PortalLinkComponent} />
                <MenuSeparator />
              </>
            ) : null}
            <MenuList
              menuSection={menuSection}
              openSubMenuLabel={openSubMenuLabel}
              hideAccentBorder={false}
              dataTestId={dataTestId ? `${dataTestId}-menulist` : undefined}
            />
            {externalLinkSections && externalLinkSections.length > 0 && (
              <ExternalLinkSections sections={externalLinkSections} />
            )}
            {extraSection && (
              <>
                <MenuSeparator />
                {extraSection}
              </>
            )}
            {languageSelectionItems && languageSelectionItems.length > 0 && (
              <>
                <MenuSeparator />
                <LanguageSelection
                  items={languageSelectionItems}
                  selected={selectedLanguage}
                  title={languageSelectionTitle}
                />
              </>
            )}
          </div>
        </nav>
      </Backdrop>
    </ServiceVariantProvider>
  ) : null;
};
