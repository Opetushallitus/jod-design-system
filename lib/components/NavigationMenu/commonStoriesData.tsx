import { JodHome } from '../../icons';
import { ExternalLinkSection } from './components/ExternalLinkSections';
import { MenuSection } from './components/MenuList';
import { DummyLangLink, DummyLink } from './storyDummyLinks';
import { LinkComponent } from './types';

export const menuSection: MenuSection = {
  title: 'Osaamispolkuni',
  linkItems: [
    {
      icon: <JodHome size={24} />,
      label: 'Osaamispolkuni',
      LinkComponent: DummyLink,
      selected: true,
    },
    {
      label: 'Kartoita tilanteesi',
      // One left in a long form as an example to remind how to use
      LinkComponent: ({ children, className, ...rest }: LinkComponent) => (
        <a href="/#" className={className} {...rest}>
          {children}
        </a>
      ),
      childItems: [
        {
          label: 'Kartoita tilanteesi',
          LinkComponent: DummyLink,
        },
        {
          label: 'Osaamisen kartoitus',
          LinkComponent: DummyLink,
        },
        {
          label: 'Kiinnostusten kartoitus',
          LinkComponent: DummyLink,
        },
        {
          label: 'Työ- ja koulutusmahdollisuudet',
          LinkComponent: DummyLink,
        },
      ],
    },
    {
      label: 'Osaamisprofiilini',
      LinkComponent: DummyLink,
      childItems: [
        {
          label: 'Osaamisprofiilini',
          LinkComponent: DummyLink,
        },
        {
          label: 'Asetukseni',
          LinkComponent: DummyLink,
        },
        {
          label: 'Suosikkini',
          LinkComponent: DummyLink,
        },
        {
          label: 'Päämääräni',
          LinkComponent: DummyLink,
        },
        {
          label: 'Osaamiseni',
          LinkComponent: DummyLink,
          childItems: [
            {
              label: 'Työpaikkani',
              LinkComponent: DummyLink,
            },
            {
              label: 'Koulutukseni',
              LinkComponent: DummyLink,
            },
            {
              label: 'Vapaa-ajan toimintoni',
              LinkComponent: DummyLink,
            },
            {
              label: 'Muut osaamiseni',
              LinkComponent: DummyLink,
            },
          ],
        },
        {
          label: 'Kiinnostuksen kohteeni',
          LinkComponent: DummyLink,
        },
        {
          label: 'Väliotsikko 2',
        },
        {
          label: 'Lorem ipsum 2',
          LinkComponent: DummyLink,
        },
      ],
    },
    {
      label: 'Väliotsikko',
    },
    {
      label: 'Lorem ipsum',
      LinkComponent: DummyLink,
    },
  ],
};

export const externalLinkSections: ExternalLinkSection[] = [
  {
    title: 'Osaamispolun sisältökokonaisuudet',
    linkItems: [
      {
        label: 'Ohjaajan osio',
        url: 'https://www.example.com',
        description: 'Ohjausalasta kiinnostuneille',
        accentColor: '#00818A',
      },
      {
        label: 'Tietopalvelu',
        url: 'https://www.example.com',
        description: 'Tietoa päätöksentekijöille',
        accentColor: '#AD4298',
      },
    ],
  },
];

export const languageSelectionItems = [
  {
    label: 'Suomeksi',
    value: 'fi',
    linkComponent: (props: LinkComponent) => <DummyLangLink href="#?lang=fi" {...props} />,
  },
  {
    label: 'På svenska',
    value: 'sv',
    linkComponent: (props: LinkComponent) => <DummyLangLink href="#?lang=sv" {...props} />,
  },
  {
    label: 'In English',
    value: 'en',
    linkComponent: (props: LinkComponent) => <DummyLangLink href="#?lang=en" {...props} />,
  },
];
