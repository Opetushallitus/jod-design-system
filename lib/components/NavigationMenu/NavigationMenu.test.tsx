import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

import { NavigationMenu, NavigationMenuProps } from './NavigationMenu';
import type { LinkComponent } from './types';

describe('NavigationMenu', () => {
  // jsdom has no support for HTMLDialogElement at the moment: https://github.com/jsdom/jsdom/issues/3294
  if (!HTMLDialogElement.prototype.showModal) {
    HTMLDialogElement.prototype.showModal = function () {
      this.open = true;
    };
  }
  if (!HTMLDialogElement.prototype.close) {
    HTMLDialogElement.prototype.close = function (returnValue: string) {
      this.open = false;
      this.returnValue = returnValue;
    };
  }

  const Link = ({ children, className }: LinkComponent) => (
    <a href="/#" className={className}>
      {children}
    </a>
  );

  const menuProps: NavigationMenuProps = {
    portalLinkLabel: 'Etusivu',
    PortalLinkComponent: Link,
    menuSection: { linkItems: [] },
    ariaCloseMenu: 'Sulje valikko',
    openSubMenuLabel: 'Avaa alivalikko',
    onClose: vi.fn(),
    externalLinkSections: [],
    languageSelectionItems: [],
    selectedLanguage: '',
    open: true,
    languageSelectionTitle: 'Käyttökieli',
    serviceVariant: 'yksilo',
    externalLinkIconAriaLabel: 'Linkki johtaa palvelun ulkopuolelle',
  };

  it('renders navigation menu with default state', () => {
    const { container } = render(<NavigationMenu {...menuProps} />);

    const navigationMenu = screen.getByRole('navigation');
    expect(navigationMenu).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('emits data-testid attributes when dataTestId is provided', () => {
    render(<NavigationMenu {...menuProps} dataTestId="navmenu" />);

    // Root wrappers (backdrop is a dialog wrapper, presence may vary in jsdom)
    expect(screen.getByTestId('navmenu-root')).toBeInTheDocument();
    expect(screen.getByTestId('navmenu-body')).toBeInTheDocument();
    // Menu list container
    expect(screen.getByTestId('navmenu-menulist')).toBeInTheDocument();
  });
});
