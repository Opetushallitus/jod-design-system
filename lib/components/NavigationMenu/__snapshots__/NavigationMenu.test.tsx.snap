// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationMenu > renders navigation menu with default state 1`] = `
<dialog
  class="ds:fixed ds:left-auto ds:right-0 ds:-z-1 ds:flex ds:backdrop:bg-black/20 ds:w-[370px] ds:max-w-full ds:max-h-screen ds:min-h-full ds:overflow-auto ds:overscroll-contain ds:hyphens-auto"
  open=""
>
  <nav
    class="ds:bg-white ds:flex ds:flex-col ds:z-100 ds:flex-1"
  >
    <div
      class="ds:px-3 ds:flex ds:flex-col ds:overflow-y-auto ds:flex-grow"
    >
      <div
        class="ds:flex ds:items-center ds:justify-end ds:my-6"
      >
        <div
          class="ds:flex ds:justify-end"
        >
          <button
            aria-label="Sulje valikko"
            class="ds:cursor-pointer ds:focus:outline-accent ds:p-3 ds:focus-visible:outline-secondary-1-dark"
          >
            <svg
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.4 19.2964L5 17.8964L10.6 12.2964L5 6.69639L6.4 5.29639L12 10.8964L17.6 5.29639L19 6.69639L13.4 12.2964L19 17.8964L17.6 19.2964L12 13.6964L6.4 19.2964Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="ds:border-l-8 ds:border-secondary-gray"
      >
        <a
          class="ds:flex ds:flex-1 ds:gap-3 ds:p-3 ds:ml-3 ds:mr-9 ds:cursor-pointer ds:group ds:rounded ds:active:text-white ds:text-black ds:hover:bg-bg-gray ds:active:bg-secondary-1-dark-2 ds:focus-visible:outline-secondary-1-dark"
          href="/#"
        >
          <span
            class="ds:text-button-md ds:group-hover:underline"
          >
            Etusivu
          </span>
        </a>
      </div>
      <hr
        class="ds:my-3 ds:h-1 ds:bg-bg-gray-2 ds:border-0"
      />
      <div>
        <ul
          class="ds:gap-2 ds:flex ds:flex-col ds:border-l-8 ds:border-secondary-1-dark"
        />
      </div>
    </div>
  </nav>
</dialog>
`;
